evap/results/tests/test_exporters.py

```diff
@@ -254,3 +255,41 @@ def test_contributor_result_export(self):
         self.assertEqual(workbook.sheets()[0].row_values(8)[2], 3.0)
         self.assertEqual(workbook.sheets()[0].row_values(10)[0], "Overall Average Grade")
         self.assertEqual(workbook.sheets()[0].row_values(10)[2], 3.25)
+
+    def test_text_answer_export(self):
+        evaluation = baker.make(Evaluation, name_en="eval_en", name_de="eval_de")
+        questions = [
+            baker.make(Question, questionnaire__type=10),
+            baker.make(Question, questionnaire__type=20),
+            baker.make(Question, questionnaire__type=30),
+        ]
+        text_answers = [
+            baker.make(TextAnswer, question=questions[0], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[1], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[2], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[2], contribution__evaluation=evaluation),
+            baker.make(TextAnswer, question=questions[0], contribution__evaluation=evaluation),
+        ]
+
+        results = defaultdict(list)
+        for text_answer in text_answers:
+            question_text = text_answer.question
```

Comment:
This is smelly, question is an actual database object, not the text of the question.

Reply:
This is how `TextAnswerExcelExporter` expects the input as of now.
Should I rather change it to a dict mapping from the tuple (question_text, question_type) to the answers, or a dict mapping from the type to (dict from question text to answers)? Or something completely different?

Reply:
I'm fine with the way you handle the data and the way the exporter want the data. My annotation was just about the local variable name `question_text`, which is misleading, since `text_answer.question` does not give you the question text, but a Question object. Since you're just using the variable in this for loop, renaming it would solve this... I guess we could just omit the variable:

```python
for text_answer in text_answers:
   results[text_answer.question].append(text_answer.answer)
```

Or am I missing something?

Reply:
You are right, this makes little sense 🤔